src/main/scala/com.snowplowanalytics.snowplow.analytics.scalasdk/decode/ValueDecoder.scala

```diff
@@ -40,42 +40,49 @@ import com.snowplowanalytics.snowplow.analytics.scalasdk.ParsingError.RowDecodin
 import com.snowplowanalytics.snowplow.analytics.scalasdk.ParsingError.RowDecodingErrorInfo._
 
 private[decode] trait ValueDecoder[A] {
-  def parse(column: (Key, String)): DecodedValue[A]
+  def parse(
+    key: Key,
+    value: String,
+    maxLength: Option[Int]
+  ): DecodedValue[A]
 }
 
 private[decode] object ValueDecoder {
   def apply[A](implicit readA: ValueDecoder[A]): ValueDecoder[A] = readA
 
-  def fromFunc[A](f: ((Key, String)) => DecodedValue[A]): ValueDecoder[A] =
+  def fromFunc[A](f: ((Key, String, Option[Int])) => DecodedValue[A]): ValueDecoder[A] =
     new ValueDecoder[A] {
-      def parse(field: (Key, String)): DecodedValue[A] = f(field)
+      def parse(
+        key: Key,
+        value: String,
+        maxLength: Option[Int]
+      ): DecodedValue[A] = f((key, value, maxLength))
     }
 
   implicit final val stringColumnDecoder: ValueDecoder[String] =
     fromFunc[String] {
-      case (key, value) =>
-        if (value.length > FIELD_SIZES.getOrElse(key.name, Int.MaxValue))
-          InvalidValue(key,
-                       value,
-                       s"Field ${key.name} longer than maximum allowed size ${FIELD_SIZES.getOrElse(key.name, Int.MaxValue)}"
-          ).asLeft
-        else if (value.isEmpty) InvalidValue(key, value, s"Field ${key.name} cannot be empty").asLeft
-        else value.asRight
+      case (key, value, Some(maxLength)) if value.length > maxLength =>
```

Comment:
Version 3.0.0 used it [on this line](https://github.com/snowplow/snowplow-scala-analytics-sdk/blob/3.0.1/src/main/scala/com.snowplowanalytics.snowplow.analytics.scalasdk/Event.scala#L254).  So it's a bit like my other comment -- we need to keep it around so that we maintain binary compatibility with previous 3.x versions.